packages/select/src/Select.tsx

```diff
@@ -334,45 +334,57 @@ export default function Select({
         return;
       }
 
-      switch (event.keyCode) {
-        case keyMap.Tab:
-        case keyMap.Escape:
-          onClose();
-          setFocusedOption(undefined);
-          break;
-        case keyMap.Enter:
-        case keyMap.Space:
-          if (open && document.activeElement !== menuButtonRef.current) {
-            // Default behavior is to use these keys to open the dropdown but we handle that manually
-            event.preventDefault();
-          }
-
-          onSelectFocusedOption(event);
-          break;
-        case keyMap.ArrowUp:
-          if (!open && document.activeElement === menuButtonRef.current) {
-            onOpen();
-          }
-
-          onFocusPreviousOption();
-          break;
-        case keyMap.ArrowDown:
-          if (!open && document.activeElement === menuButtonRef.current) {
-            onOpen();
-          }
-
-          onFocusNextOption();
-          break;
+      const isFocusInMenu = listMenuRef.current?.contains(
+        document.activeElement,
+      );
+      const isFocusOnButton = menuButtonRef.current?.contains(
+        document.activeElement,
+      );
+      const isFocusInComponent = isFocusOnButton || isFocusInMenu;
+
+      // We only respond to keypresses if the focus is in the component
+      if (isFocusInComponent) {
+        switch (event.key) {
```

Comment:
Changing the switch parameter since `event.keyCode` [is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode)

Reply:
Is this what caused the issue? We've kept using keyCode because `KeyboardEvent.key` is only partially supported in IE 11, and keyCode has consistent browser support currently. I don't think _most_ teams support IE 11, but it seems like something we should confirm before making this change.

Reply:
It's not the source of the issue, no, but I was getting IDE deprecation warnings. Didn't realize IE had only parital support. Looking into it, it seems IE only has issues regarding the `AltGraph` key, but I may be misreading https://caniuse.com/keyboardevent-key

Reply:
I'll revert if that's something we want to stick with

Reply:
If it is in fact only AltGraph, then it could be fine though I tend to avoid APIs with inconsistent behavior even if we don't _think_ we use it or will use it. We should do some research to understand the current browser support we offer in all of our properties before moving to `KeyboardEvent.key`, since the `keyCode` property is still widely supported and is consistent.

Let's keep this PR simple so we can get the fix in, and evaluate this separately. If we do make this update, we should make it globally.